frontend/src/metabase/dashboard/components/AutoApplyFilterToast/AutoApplyFilterToast.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useMemo } from "react";
+import { t } from "ttag";
+
+import { useSelector, useDispatch } from "metabase/lib/redux";
+import {
+  getAutoApplyFiltersToastStateName,
+  getDashboardId,
+  getIsAutoApplyFilters,
+  getParameterValues,
+} from "metabase/dashboard/selectors";
+import {
+  dismissAutoApplyFiltersToast,
+  saveDashboardAndCards,
+  setDashboardAttributes,
+  setNeverShowAutoApplyFiltersToast,
+  setReadyForAutoApplyFiltersToast,
+} from "metabase/dashboard/actions";
+import { addUndo } from "metabase/redux/undo";
+import { ToasterButton } from "metabase/components/Toaster/Toaster.styled";
+
+export default function AutoApplyFilterToast() {
+  const isAutoApplyFilters = useSelector(getIsAutoApplyFilters);
+  const parameterValues = useSelector(getParameterValues);
+  const autoApplyFiltersToastStateName = useSelector(
+    getAutoApplyFiltersToastStateName,
+  );
+  const dashboardId = useSelector(getDashboardId);
+
+  const hasParameterValues = useMemo(
+    () =>
+      Object.values(parameterValues).some(parameterValue =>
+        Array.isArray(parameterValue)
+          ? parameterValue.length > 0
+          : parameterValue != null,
+      ),
+    [parameterValues],
+  );
+  const isReadyForToast = isAutoApplyFilters && hasParameterValues;
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (isReadyForToast) {
+      dispatch(setReadyForAutoApplyFiltersToast());
```

Comment:
actions dispatched within this components follows the logic in this state machine
```mermaid
stateDiagram-v2
never: Never show toast
ready: Ready for toast
loading: Loading cards
timedOut: Timed out
shown: Toast visible

never --> ready: ready condition
ready --> never: never condition

ready --> loading: load()

loading --> timedOut: timed out (15s)
loading --> loading: load()
loading --> never: never condition
loading --> ready: loaded

timedOut --> loading: load()
timedOut --> shown: loaded
timedOut --> never: never condition

shown --> never: never condition \n(turn-off auto-apply filters)
shown --> ready: dismiss toast

note left of never
  Never condition.
  When`auto_apply_filters: false`
  OR has no `parameterValues`
end note

note left of ready
  Ready condition.
  When`auto_apply_filters: true`
  AND has `parameterValues`
end note
```